manim/scene/scene.py

```diff
@@ -596,85 +596,47 @@ def get_moving_and_static_mobjects(self, animations):
 
         Parameters
         ----------
-        *args : Animation or method of a mobject, which is followed by that method's arguments
+        *args : Union[:class:`Animation`, :class:`AnimationBuilder`]
+            Animations and AnimationBuilders to be played.
 
-        **kwargs : any named arguments like run_time or lag_ratio.
+        **kwargs
+            Configuration for the call to play().
 
         Returns
         -------
-        list : list of animations with the parameters applied to them.
+        List[:class:`Animation`]
+            List of Animations to be played.
         """
         animations = []
-        state = {
-            "curr_method": None,
-            "last_method": None,
-            "method_args": [],
-        }
-
-        def compile_method(state):
-            if state["curr_method"] is None:
-                return
-            mobject = state["curr_method"].__self__
-            if state["last_method"] and state["last_method"].__self__ is mobject:
-                animations.pop()
-                # method should already have target then.
-            else:
-                mobject.generate_target()
-            #
-            if len(state["method_args"]) > 0 and isinstance(
-                state["method_args"][-1], dict
-            ):
-                method_kwargs = state["method_args"].pop()
-            else:
-                method_kwargs = {}
-            state["curr_method"].__func__(
-                mobject.target, *state["method_args"], **method_kwargs
-            )
-            animations.append(MoveToTarget(mobject))
-            state["last_method"] = state["curr_method"]
-            state["curr_method"] = None
-            state["method_args"] = []
-
         for arg in args:
-            if isinstance(arg, Animation):
-                compile_method(state)
+            if isinstance(arg, _AnimationBuilder):
+                animations.append(self.build_animation(arg))
+            elif isinstance(arg, Animation):
                 animations.append(arg)
-            elif inspect.ismethod(arg):
-                compile_method(state)
-                state["curr_method"] = arg
-            elif state["curr_method"] is not None:
-                state["method_args"].append(arg)
-            elif isinstance(arg, Mobject):
-                raise ValueError(
-                    """
-                    I think you may have invoked a method
-                    you meant to pass in as a Scene.play argument
-                    """
-                )
             else:
-                raise ValueError("Invalid play arguments")
-        compile_method(state)
+                raise ValueError(f"Invalid play argument {arg}")
 
         for animation in animations:
-            # This is where kwargs to play like run_time and rate_func
-            # get applied to all animations
-            animation.update_config(**kwargs)
-
+            animation.__dict__.update(**kwargs)
```

Comment:
I don't think manually touching `__dict__` is a great idea, especially because in this instance it'll just ignore properties/descriptors in general. I'd suggest using `setattr` instead
```suggestion
            for attr, value in kwargs.items():
                setattr(animation, attr, value)
```

Reply:
Oh and if you choose to stay with touching `__dict__`, you can just do `animation.__dict__.update(kwargs)`, no need to forward them.

Reply:
This is actually just copy pasting the definition of `update_config` since it was a single line function that was only called in one place. I can still do the changes if you still want to.

Reply:
Yeah, I'd appreciate the changes just so it works with whatever properties and such an animation might have, and for readability as `__dict__` isn't the most well-known thing.

Reply:
done.